### ISSUE TITLE
Fine-tune InceptionV3, error in documentation

### ISSUE THREAD

**Issue by marthall** [fchollet/keras#6910]
Hi,

I believe there is an error in the example called  [Fine-tune InceptionV3 on a new set of classes](https://github.com/fchollet/keras/blob/master/docs/templates/applications.md#fine-tune-inceptionv3-on-a-new-set-of-classes).

Specifically, it says:
```python
# we chose to train the top 2 inception blocks, i.e. we will freeze
# the first 172 layers and unfreeze the rest:
for layer in model.layers[:172]:
   layer.trainable = False
for layer in model.layers[172:]:
   layer.trainable = True
```

However, let us consider the following code:
```python
from keras.applications.inception_v3 import InceptionV3
from keras.utils import plot_model

model = InceptionV3(include_top=False)
plot_model(model, to_file='model.png')

for i, layer in enumerate(base_model.layers):
   print(i, layer.name)
```
The `model.layers[172:]` does not correspond to the last two inception blocks, more like 4 and a half. I believe this is an error in the documentation. Can anyone confirm that my process is correct?

**Comment by myutwo150** [comment]
Yes, mixed8 is now the 248th layer.
So it should be 249 instead of 172 in the doc.

**Comment by preetskhalsa97** [comment]
Hey!
I was following the[ documentation](https://github.com/keras-team/keras/blob/master/docs/templates/applications.md#fine-tune-inceptionv3-on-a-new-set-of-classes) on how to finetune on InceptionV3.
It says:
for layer in base_model.layers:
   layer.trainable = False# compile the model (should be done *after* setting layers to non-trainable)
model.compile(optimizer='rmsprop', loss='categorical_crossentropy')# train the model on the new data for a few epochs
model.fit_generator(...)# at this point, the top layers are well trained and we can start fine-tuning
# convolutional layers from inception V3. We will freeze the bottom N layers
# and train the remaining top layers.# let's visualize layer names and layer indices to see how many layers
# we should freeze:
print (model.summary())
for i, layer in enumerate(base_model.layers):
  print(i, layer.name)# we chose to train the top 2 inception blocks, i.e. we will freeze
# the first 249 layers and unfreeze the rest:
for layer in model.layers[:249]:
  layer.trainable = False
for layer in model.layers[249:]:
  layer.trainable = True

print (model.summary())

However, on printing the model summary, the trainable parameters are same in number both the instances where model.summary() is printed. In that case, what extra are we doing while finetuning?

**Comment by Amritpal-001** [comment]
This was helpful. So we always try to freeze a model till it's mixed/concatenate layer????
